spec/presenters/admin_collection_presenter_spec.rb

```diff
@@ -18,73 +18,83 @@
+  let(:unit_name) { "Default Unit" }
   let(:description) { "The long form description of this collection." }
   let(:managers) { [manager.to_s] }
-  let(:editors) { [editor.to_s, FactoryBot.create(:manager)] }
+  let(:editors) { [editor.to_s, FactoryBot.create(:manager).to_s] }
   let(:depositors) { [depositor.to_s] }
   let(:manager) { FactoryBot.create(:manager) }
   let(:editor) { FactoryBot.create(:user) }
   let(:depositor) { FactoryBot.create(:user) }
-  let(:solr_doc) do
-    SolrDocument.new(
-      id: id,
-      "name_ssi": name,
-      "unit_ssi": unit,
-      "description_tesim": [description],
-      "edit_access_person_ssim": managers + editors,
-      "read_access_person_ssim": depositors,
-      "collection_managers_ssim": managers
-    )
-  end
+  let(:unit) { FactoryBot.create(:unit, name: unit_name) }
+  let(:collection) { FactoryBot.create(:collection, id: id, name: name, unit: unit, description: description, managers: managers, editors: editors, depositors: depositors) }
+  let(:solr_doc) { SolrDocument.new(collection.to_solr) }
   subject(:presenter) { described_class.new(solr_doc) }
 
   it 'provides getters for descriptive fields' do
     expect(presenter.id).to eq id
     expect(presenter.name).to eq name
-    expect(presenter.unit).to eq unit
+    expect(presenter.unit).to eq unit_name
     expect(presenter.description).to eq description
   end
 
   it 'provides access control information' do
-    expect(presenter.managers).to eq managers
-    expect(presenter.editors).to eq editors
-    expect(presenter.depositors).to eq depositors
+    expect(presenter.managers).to match_array managers
+    expect(presenter.editors).to match_array editors
+    expect(presenter.depositors).to match_array depositors
   end
 
   describe '#as_json' do
     subject(:presenter_json) { presenter.as_json({}) }
     it 'returns json' do
       expect(presenter_json[:id]).to eq id
       expect(presenter_json[:name]).to eq name
-      expect(presenter_json[:unit]).to eq unit
+      expect(presenter_json[:unit]).to eq unit_name
       expect(presenter_json[:description]).to eq description
       expect(presenter_json[:object_count]).to be_present
-      expect(presenter_json[:roles][:managers]).to eq managers
-      expect(presenter_json[:roles][:editors]).to eq editors
-      expect(presenter_json[:roles][:depositors]).to eq depositors
+      expect(presenter_json[:roles][:managers]).to match_array managers
+      expect(presenter_json[:roles][:editors]).to match_array editors
+      expect(presenter_json[:roles][:depositors]).to match_array depositors
     end
   end
 
   describe 'abilities' do
-    subject(:ability) { Ability.new(user, {}) }
+    subject(:ability) { Ability.new(User.where(Devise.authentication_keys.first => user).first) }
 
     context 'when manager' do
       let(:user) { manager }
 
-      it { is_expected.to be_able_to(:destroy, presenter) }
+      it { is_expected.not_to be_able_to(:destroy, presenter) }
```

Comment:
```suggestion
      it { is_expected.to be_able_to(:destroy, presenter) }
```

I think this got changed accidentally. Though the fact it is passing while the inherited from unit test with the opposite check is also passing indicates there is something wrong with the ability definitions or that the `be_able_to` check or negation of that check is not functioning as expected.